Winleafs.Wpf/Api/Events/SpotifyEventTrigger.cs

```diff
@@ -0,0 +1,47 @@
+﻿using NLog;
+using System.Threading.Tasks;
+using System.Timers;
+
+namespace Winleafs.Wpf.Api.Events
+{
+    /// <summary>
+    /// Event trigger that is activated when a Spotify playlist is played
+    /// </summary>
+    public class SpotifyEventTrigger : EventTriggerBase
+    {
+        private readonly EventTriggersCollection _eventTriggersCollection;
+        private readonly string _playlistId;
+
+        public SpotifyEventTrigger(EventTriggersCollection eventTriggersCollection, Models.Models.Scheduling.Triggers.SpotifyEventTrigger spotifyEventTrigger)
+            : base(spotifyEventTrigger.Brightness, spotifyEventTrigger.EffectName, spotifyEventTrigger.Priority)
+        {
+            _eventTriggersCollection = eventTriggersCollection;
+            _playlistId = spotifyEventTrigger.PlaylistId;
+
+            var processCheckTimer = new Timer(60000);
+            processCheckTimer.Elapsed += CheckProcess;
+            processCheckTimer.AutoReset = true;
+            processCheckTimer.Start();
+        }
+
+        private void CheckProcess(object source, ElapsedEventArgs e)
+        {
+            Task.Run(() => CheckPlaylistAsync());
+        }
+
+        /// <summary>
+        /// Checks if a playlist is being played in Spotify then execute TryStartEffect(), else stop the effect
+        /// </summary>
+        private async Task CheckPlaylistAsync()
```

Comment:
Why not make the API request in there? Feels weird that this is based off two timers that could be desynced.

Reply:
Simple: if there are multiple spotify triggers, then we would make multiple checks a minute, while a user can only play 1 playlist at a time. So the requests are done in a static class to decrease the amounts of requests to our server and Spotify servers as well.

Reply:
Okay I agree